primitives/test-utils/src/lib.rs

```diff
@@ -67,6 +96,29 @@ pub fn make_justification_for_header<H: HeaderT>(
 	}
 }
 
+fn generate_chain<H: HeaderT>(fork_id: u8, depth: u32, ancestor: &H) -> Vec<H> {
+	let mut headers = vec![ancestor.clone()];
+
+	for i in 1..depth {
+		let parent = &headers[(i - 1) as usize];
+		let (hash, num) = (parent.hash(), *parent.number());
+
+		let mut header = test_header::<H>(num + One::one());
+		header.set_parent_hash(hash);
+
+		// Modifying the digest so headers at the same height but in different forks have different
+		// hashes
+		let digest = header.digest_mut();
+		*digest = sp_runtime::Digest {
```

Comment:
We could use authority id as well (would be a bit more like babe :)).

Reply:
Haha, we could, but I don't really see the benefits in this case. We need the number of forks anyways, might as well use that since it's easy